### PR TITLE
Cmr 514

### PR DESCRIPTION
**Pull request by dseelig**

### CODE REVIEW

File: src/main/java/de/qucosa/component/oaipmh/model/Format.java [hunk 1]
```diff
@@ -31,11 +29,14 @@
...
+    @JsonProperty("schemaurl")
+    private String schemarl;
+
+    @JsonProperty("namespace")
+    private String namespace;
 
-    @JsonProperty("dissType")
-    private String dissType;
+    @JsonProperty("deleted")
+    private boolean deleted; private String schemaurl;
```
Comment (claussni) [inline src/main/java/de/qucosa/component/oaipmh/model/Format.java]: Missing a new line here?

File: src/main/java/de/qucosa/model/DissTerms.java [hunk 2]
```diff
@@ -16,18 +16,12 @@
 
 package de.qucosa.model;
 
-import com.fasterxml.jackson.annotation.JsonAutoDetect;
-import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.*;
```
Comment (claussni) [inline src/main/java/de/qucosa/model/DissTerms.java]: Please configure your IDE to always expand imports.

File: src/main/java/de/qucosa/model/DissTerms.java [hunk 3]
```diff
@@ -51,8 +48,34 @@
...
+
+        if (config instanceof String) {
+            this.config = getClass().getResourceAsStream((String) config);
+
+            if (this.config == null) {
+
+                try {
+                    this.config = new FileInputStream(new File((String) config));
+                } catch (FileNotFoundException e) {
+                    logger.error("dissemination-config.json is not found.", e);
```
Comment (claussni) [inline src/main/java/de/qucosa/model/DissTerms.java]: Isn't it a fatal error if the configuration file cannot be found? How can processing continue? There should be a an program aborting exception traveling out to the caller.

File: src/main/java/de/qucosa/model/SetsConfig.java [hunk 4]
```diff
@@ -79,63 +105,51 @@ public Set getSetObject(String setSpec) {
...
 
             try {
-                sets = om.readValue(setSpecs, om.getTypeFactory().constructCollectionType(List.class, Set.class));
-            } catch (JsonParseException e) {
-                e.printStackTrace();
-            } catch (JsonMappingException e) {
-                e.printStackTrace();
+                sets = om.readValue(stream, om.getTypeFactory().constructCollectionType(List.class, Set.class));
             } catch (IOException e) {
                 e.printStackTrace();
```
Comment (claussni) [inline src/main/java/de/qucosa/model/SetsConfig.java]: Please don't print stack traces. Log something, or throw an exception.

File: src/main/java/de/qucosa/model/DissTerms.java [hunk 5]
```diff
@@ -212,18 +234,17 @@ private DissTermsDao dao() {
...
+        public DissTermsDao(InputStream stream) {
             ObjectMapper om = new ObjectMapper();
-            File file = new File(configPath + "/dissemination-config.json");
 
             try {
-                dissTerms = om.readValue(Files.readAllBytes(Paths.get(file.getAbsolutePath())), DissTerms.class);
+                dissTerms = om.readValue(stream, DissTerms.class);
             } catch (IOException e) {
-                e.printStackTrace();
                 logger.debug("dissemination-conf parse failed.");
```
Comment (claussni) [inline src/main/java/de/qucosa/model/DissTerms.java]: Why log.debug? Isn't this a fatal error?

File: src/main/java/de/qucosa/model/SetsConfig.java [hunk 6]
```diff
@@ -53,10 +53,36 @@
...
+
+        if (config instanceof String) {
+            this.config = getClass().getResourceAsStream((String) config);
+
+            if (this.config == null) {
+
+                try {
+                    this.config = new FileInputStream(new File((String) config));
+                } catch (FileNotFoundException e) {
+                    logger.error("list-set-conf.json is not found.", e);
```
Comment (claussni) [inline src/main/java/de/qucosa/model/SetsConfig.java]: You're getting a JSON imput stream here. You actually don't know the that it came from a file and what this files name was.

File: src/main/java/de/qucosa/model/SetsConfig.java [hunk 7]
```diff
@@ -53,10 +53,36 @@
...
-        this.configPath = configPath;
+    public <T> SetsConfig(@JsonProperty("config") T config) {
+
+        if (config instanceof String) {
+            this.config = getClass().getResourceAsStream((String) config);
+
+            if (this.config == null) {
+
+                try {
+                    this.config = new FileInputStream(new File((String) config));
```
Comment (claussni) [inline src/main/java/de/qucosa/model/SetsConfig.java]: Why are you expecting a String here? `config` is of type `T`. What is `T`?

Comment (dseelig) [reply src/main/java/de/qucosa/model/SetsConfig.java]: This problem will be fixed in https://jira.slub-dresden.de/browse/CMR-518

**Pull request merged**